pandas/core/reshape/concat.py

```diff
@@ -585,6 +624,66 @@ def _maybe_check_integrity(self, concat_index: Index):
                     "{overlap!s}".format(overlap=overlap)
                 )
 
+    def _items_overlap_with_suffix(self, objs, suffixes):
+        """
+        Adding suffix for items if there is overlapping situation.
+
+        Be aware that `objs` can be either DataFrame-like or Index-like given
+        if `self._is_series` is True or False.
+        """
+        if self._is_series:
+
+            # when _is_series is True, objs are actually column Index
+            overlap_cols = list(objs)
+        else:
+            overlap_cols = chain.from_iterable([obj.columns for obj in objs])
+        to_rename = [col for col, cnt in Counter(overlap_cols).items() if cnt > 1]
+
+        if len(to_rename) == 0 or suffixes is None:
+            return objs
+
+        if not isinstance(suffixes, tuple):
```

Comment:
Should this `isinstance` check maybe be moved to the start of the function? Might be clearer.

Reply:
i thought so as well, and then decided to leave it here because of two reasons:
1. this will only be used if there is overlapping in column names.
2. the default is `None`, so default is used, then directly return original `objs` without post-processing and checking below. 
Does it make any sense? I am very open to suggestions!